src/deepsparse/loggers/helpers.py

```diff
@@ -11,15 +11,74 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-import re
-from typing import Any, Optional, Tuple
-
 """
 Helpers functions for logging
 """
 
-__all__ = ["match_and_extract"]
+import importlib
+import os.path
+import re
+from typing import Any, Callable, Optional, Tuple
+
+import numpy
+
+import deepsparse.loggers.metric_functions.built_ins as built_ins
+import torch
+
+
+__all__ = ["match_and_extract", "get_function_and_function_name"]
+
+
+def get_function_and_function_name(
+    function_identifier: str,
+) -> Tuple[Callable[[Any], Any], str]:
+    """
+    Parse function identifier and return the function as well as its name
+
+    :param function_identifier: Can be one of the following:
+
+        1. framework function, e.g.
+            "torch.mean" or "numpy.max"
+
+        2. built-in function, e.g.
+            "function_name"
+            note: function needs to be available in the module
+            with built-in functions, see the imports above)
+
+        3. user-defined function in the form of
+           '<path_to_the_python_script>:<function_name>', e.g.
+           "{...}/script_name.py:function_name"
+
+    :return: A tuple (function, function name)
+    """
+
+    if function_identifier.startswith("torch."):
+        func_name = function_identifier.split(".")[1]
+        return getattr(torch, func_name), func_name
+
+    if function_identifier.startswith("numpy.") or function_identifier.startswith(
+        "np."
+    ):
+        func_name = function_identifier.split(".")[1]
+        return getattr(numpy, func_name), func_name
```

Comment:
accessing only `function_identifier.split(".")[1]` isn't quite enough because there may be nested functions. for example `torch.linalg.l2` would return `torch.linalg`